tests/containers/container_diff.pm

```diff
@@ -21,16 +21,39 @@ sub run {
 
     zypper_call("install container-diff") if (script_run("which container-diff") != 0);
 
-    my ($untested_images, $released_images) = get_suse_container_urls();
+    my ($testing_images, $released_images) = get_suse_container_urls();
     # container-diff
-    for my $i (@{$untested_images}) {
-        my $image_file = $untested_images->[$i] =~ s/\/|:/-/gr;
+    for my $i (@{$testing_images}) {
+        my $image_file = $testing_images->[$i] =~ s/\/|:/-/gr;
         my $container_diff_results = "/tmp/container-diff-$image_file.txt";
-        assert_script_run("docker pull $untested_images->[$i]", 360);
+        assert_script_run("docker pull $testing_images->[$i]", 360);
         assert_script_run("docker pull $released_images->[$i]", 360);
-        assert_script_run("container-diff diff daemon://$untested_images->[$i] daemon://$released_images->[$i] --type=rpm --type=file --type=size > $container_diff_results", 300);
+        assert_script_run("container-diff diff daemon://$testing_images->[$i] daemon://$released_images->[$i] --type=rpm --type=file --type=size > $container_diff_results", 300);
         upload_logs("$container_diff_results");
-        ensure_container_rpm_updates("$container_diff_results");
+        # Check if the new container is not an empty update
+        assert_script_run("container-diff diff daemon://$testing_images->[$i] daemon://$released_images->[$i] --type=file > $container_diff_results");
+        # The output from `container-diff diff ...` looks like the following:
+        # -----File-----
+        #
+        # These entries have been added to registry.opensuse.org/opensuse/tumbleweed:
+        # FILE                                                                                  SIZE
+        # /etc/exports.d                                                                        0
+        # /etc/issue.net                                                                        58B
+        # /etc/motd.d                                                                           0
+        # -- snip --
+        #
+        # If the diff is empty, the output looks like the following:
+        # -----File-----
+        #
+        # These entries have been added to registry.opensuse.org/opensuse/tumbleweed: None
+        #
+        # These entries have been deleted from registry.opensuse.org/opensuse/tumbleweed: None
+        #
+        # These entries have been changed between registry.opensuse.org/opensuse/tumbleweed and registry.opensuse.org/opensuse/tumbleweed: None
+        record_info("Empty container update (no changes)") if (script_run("grep -v 'FILE' $container_diff_results", 300) != 0);
+
+        # Ensure there are no pending updates in the container
+        assert_script_run("docker run -t --rm $testing_images->[$i] /bin/bash -c 'zypper ref && zypper -n up -D | grep \"Nothing to do.\"'", timeout => 300);
```

Comment:
IMO this should be done in a different module, alongside the other zypper tests

Reply:
@jlausuch You have a better overview if this check should be moved elsewhere, I was only fixing the occurring failure. Move it or leave it?

Reply:
Yes, this looks more like pure "image test" ensuring that a fresh built image is up to date completely and not a container-diff thingy.. I would remove it from here.

Reply:
Removed it